File: leetcode/golang/LCR_141_fan_zhuan_lian_biao_lcof.go
type ListNode struct {
     Val int
     Next *ListNode
}

func trainningPlan(head *ListNode) *ListNode {
    var tmp = &ListNode{}
    node := head
    tmpHead := head
    for tmpHead != nil {
        node = tmpHead.Next
        tmpHead.Next = tmp.Next
        tmp.Next = tmpHead
        tmpHead = node
    }
    return tmp.Next
}

File: leetcode/c++/LCR_141_fan_zhuan_lian_biao_lcof.cpp
// https://leetcode.cn/problems/fan-zhuan-lian-biao-lcof/description

#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* trainningPlan(ListNode* head) {
        ListNode* tmp = new ListNode();
        ListNode* node = head;
        while (head) {
            node = head->next;
            head->next = tmp->next;
            tmp->next = head;
            head = node;
        }
        return tmp->next;
    }
};

// 时间复杂度为O(n)，空间复杂度为O(1)